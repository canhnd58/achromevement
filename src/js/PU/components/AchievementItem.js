import React from 'react';

/* const Tiers = Object.freeze({
  BRONZE: 1,
  SILVER: 2,
  GOLD: 3,
}); */

const displayTier = t =>
  ({
    1: 'BRONZE',
    2: 'SILVER',
    3: 'GOLD',
  }[t]);

const AchievementItem = ({
  title,
  description,
  goal,
  hidden,
  done,
  earned,
  tier,
}) => (
  <div>
    <div className="title">
      <b>{title}</b> ({displayTier(tier)})
    </div>
    {(!hidden || earned) && (
      <>
        <div className="description">
          <i>{description}</i>
        </div>
        <div className="progress">
          Progress: {done}
          &#47;
          {goal}
        </div>
      </>
    )}
    <hr />
  </div>
);

export default AchievementItem;
