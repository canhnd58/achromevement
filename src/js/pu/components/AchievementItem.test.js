import React from 'react';
import { shallow } from 'enzyme';

import AchievementItem from './AchievementItem';

const defaultProps = {
  title: 'No Title',
  description: 'No description',
  goal: [2, 7, 30],
  hidden: false,
  done: 3,
  earned: false,
  tier: 2,
};

describe('AchievementItem', () => {
  test('render correctly', () => {
    const component = shallow(<AchievementItem {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });

  test('do not show description with "hidden" prop', () => {
    const component = shallow(
      <AchievementItem {...defaultProps} hidden={true} />
    );
    expect(component).toMatchSnapshot();
  });

  test('show hidden description with "earned" prop', () => {
    const component = shallow(
      <AchievementItem {...defaultProps} hidden={true} earned={true} />
    );
    expect(component).toMatchSnapshot();
  });
});
