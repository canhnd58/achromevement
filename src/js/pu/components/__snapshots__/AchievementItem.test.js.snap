// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementItem do not show description with "hidden" prop 1`] = `
<div>
  <div
    className="title"
  >
    <b>
      No Title
    </b>
     (
    SILVER
    )
  </div>
</div>
`;

exports[`AchievementItem render correctly 1`] = `
<div>
  <div
    className="title"
  >
    <b>
      No Title
    </b>
     (
    SILVER
    )
  </div>
  <div
    className="description"
  >
    <i>
      No description
    </i>
  </div>
  <div
    className="progress"
  >
    Progress: 
    3
    /
    2
    7
    30
  </div>
</div>
`;

exports[`AchievementItem show hidden description with "earned" prop 1`] = `
<div>
  <div
    className="title"
  >
    <b>
      No Title
    </b>
     (
    SILVER
    )
  </div>
  <div
    className="description"
  >
    <i>
      No description
    </i>
  </div>
  <div
    className="progress"
  >
    Progress: 
    3
    /
    2
    7
    30
  </div>
</div>
`;
